Yahoo weather provider fixed

grueni75 opened the pull request:
Hello benjymous.

are you still working on the community edition? I didn't saw any updates from your side lately. Nevertheless, yahoo has shut down the free of charge placefinder API. I fixed the code to use the non commercial yahoo tables API. Here is my commit comment:

Changed code to use the yahoo tables API
Lookup is limited to 2000 queries per day, but should be enough if each metawatch user gets an own internet IP
But may be better to add an other free weather provider for the long term

Will you update the community edition on the Play Store or did you stop working on Metawatch?

Bye

Matthias

Commit:
Free of charge yahoo place finder API has been shut down
Changed code to use the yahoo tables API
Lookup is limited to 2000 queries per day, but should be enough if each metawatch user gets an own internet IP
But may be better to add an other free weather provider for the long term

jbtbnl commented:
A workaround is to use Weather Underground instead of Yahoo weather.
You only need to sign up for an API key.

grueni75 commented:
I know. But it's not working anymore here in Germany.

sdemills commented:
Have you tried contacting Wunderground in Germany to find out why it's not working? It may simply be a fault. I have a sophisticated weather station which provides a direct real time feed to Wunderground, part of a huge worldwide community of such people. I'd be surprised and disconcerted to hear that Wunderground had withdrawn their service in such s major world power as Germany. So now I'm keen to know what's happening.

Commit:
new feature: rate test added for firmware debugging

Commit:
enhancement: better state handling after watch reconnects

Commit:
build updates

Commit:
bug fix: if an app was removed from the idle page list, the first press of the "next idle screen" button was ignored

Commit:
improvement: more pretty day positioning if widget separator is active

Commit:
new feature: regular expression for extracting the short location (e.g., room numnber) from a long location description

Commit:
new feature: debug log control and download (requires grueni75's BTStack based firmware)

Commit:
bug fix: phone action screen was not left if phone call was over

Commit:
bug fix: phone call action screen was sometimes garbled if outgoing call was initiaed

Commit:
enhancement: better handling of the watch state / screen update after a reconnect

Commit:
enhancement: better drawing of calendar and weather widget if separator line is activated

Commit:
version up

Commit:
APK file added for other to download

Commit:
bug fix: regexp for lotus notes room descriptions

Commit:
new feature: rate test with random bitmap

Commit:
new feature: selectable default view for calendar app

Commit:
new feature: location is displayed in agenda view of calendar app

Commit:
version up / new build

Commit:
new file ignroed

Commit:
New feature: Use text extracted from content view if notification has no ticker text
This will display notifications from apps such as Google Keep on the watch

Commit:
Removed permissions only granted to system apps
Fixed missing music meta information on Android 4.3 / GN9005

Commit:
bug fix: music player was not re-displayed correctly after mode timeout

Commit:
bug fix: gmail notification now working with the new multiple inbox labels

Commit:
bug fix: phone call actions were not dismissed correctly

Commit:
bug fix: state was not correctly restored after mode timeout

Commit:
new feature: whatsapp and hangout are notified as SMS with full message contents
bug fix: goggle now and keep were not displayed correctly on Android 4.3

Commit:
new build

Commit:
misc: format changes
new feature: recover watch state if it does not match app state

Commit:
misc: format changes
enhancement: already shown notification detection now based on history per app
enhancement: complete text from content view is used if ticker text is empty

Commit:
new build

Commit:
new build

Commit:
bug fix: if watch type was not detected yet and an notification arrived, the notification handler got stucked

Commit:
enhancement: state monitor added that restores watch lcd buffer if it is not matching the mode
new feature: widget that shows weight from withings body scale
new feature: widget that shows environment data like temperature, humidity and pressure
enhancement: minor changes in weather widget layout to match the environment widget

Commit:
enhancement: withings widget is hidden if no data available

Commit:
EnvironmentSensorWidget: prev pressure value is stored in shared preferences such that the tendency indication also works after restart
WithingsWidget: values are remembered in shared prefs such that last weight is also shwon if withings server is not available

Commit:
new feature: intent for starting metawatch service

Commit:
new feature: Combining multiple regexp for extracting location from appointment

Commit:
new build

Commit:
bug fix: if a widget row is empty, wrong seperator row was drawn

Commit:
bug fix: pressure tendency was not shown if pressure does not always change within one hour

Commit:
new release

Commit:
improvement: pressure tendency already after 1 hPa change
improvement: pressure tendency reset after 3 hours of unchanged pressure

Commit:
new build

Commit:
switched to kit kat
improvement: idle screen is updated with last app bitmap if app is started
new build

Commit:
bug fix: switch to screen of app if app starts

Commit:
bug fix: calendars were hidden if same calendar is visible under different accounts

Commit:
new build

Commit:
bug fix: call controls did not work anymore

Commit:
new android build

Commit:
improvement: time to end of meeting is now shown in minutes instead of milliseconds

Commit:
New android build

Commit:
bug fix: fixed crash if no location is available

Commit:
new android build

Commit:
bug fix: idle pages are only resetted when settings are changed

Commit:
new android build

Commit:
bug fix: calendar app did not reload calendar if app is running for more than one day

Commit:
bug fix: calendar app did not reload calendar if app is running for more than one day

Commit:
bug fix: calendar app did not reload calendar if app is running for more than one day

Commit:
Revert "bug fix: calendar app did not reload calendar if app is running for more than one day"

This reverts commit 7bd441b2eaf232224e2107c946a9d621b7271379.

Commit:
Revert "bug fix: calendar app did not reload calendar if app is running for more than one day"

This reverts commit 1b0d25047b362e73ba7478998fe7aa3ababd25b1.

Commit:
bug fix: refresh calendar app if day changes

Commit:
Merge branch 'master' of git@github.com:grueni75/MWM-for-Android.git

Conflicts:
	bin/MetaWatch.apk

Commit:
improvement: changed order of hangouts and whatsapp messages (latest first)
